File: LeetCode-Problems/Easy/Binary-To-Integer-Linked-List.py
# Python
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def getDecimalValue(self, head):
        """
        :type head: ListNode
        :rtype: int
        """
        count = ""
        current = head
        while(current != None):
            count += str(current.val)
            current = current.next
        return int(count, 2)

File: LeetCode-Problems/Easy/Binary-To-Integer-Linked-List.java
//Java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        String count = "";
        ListNode current = head;
        while (current != null){
            count += Integer.toString(current.val);
            current = current.next;
        }
        return Integer.parseInt(count, 2);
        
    }
}
